localstack/services/kms/provider.py

```diff
@@ -133,182 +159,336 @@ def __init__(self, message=None):
+        kwargs = {"NextMarker": next_token, "Truncated": True} if next_token else {}
+
+        return ListGrantsResponse(Grants=page, **kwargs)
+
+    # Honestly, this is a mess in AWS KMS. Hashtag "do we follow specifications that are a pain to customers or do we
+    # diverge from AWS and make the life of our customers easier?"
+    #
+    # Both RetireGrant and RevokeGrant operations delete a grant. The differences between them are described here:
+    # https://docs.aws.amazon.com/kms/latest/developerguide/grant-manage.html#grant-delete
+    # Essentially:
+    # - Permissions to RevokeGrant are controlled through IAM policies or through key policies, while permissions to
+    # RetireGrant are controlled by settings inside the grant itself.
+    # - A grant to be retired can be specified by its GrantToken or its GrantId/KeyId pair. While revoking grants can
+    # only be done with a GrantId/KeyId pair.
+    # - For RevokeGrant, KeyId can be either an actual key ID, or an ARN of that key. While for RetireGrant only key
+    # ARN is accepted as a KeyId.
+    #
+    # We currently do not model permissions for retirement and revocation of grants. At least not in KMS,
+    # maybe IAM in LocalStack has some modelling though. We also accept both key IDs and key ARNs for both
+    # operations. So apart from RevokeGrant not accepting GrantToken parameter, we treat these two operations the same.
+    def _delete_grant(
+        self, store: KmsStore, grant_id: str = None, key_id: str = None, grant_token: str = None
+    ):
+        if grant_token:
+            if grant_token not in store.grant_tokens:
+                raise NotFoundException(f"Unable to find grant token {grant_token}")
+            grant_id = store.grant_tokens[grant_token]
+            # Do not really care about the key ID if a grant is identified by a token. But since a key has to be
+            # validated when a grant is identified by GrantId/KeyId pair, and since we want to use the same code in
+            # both cases - when we have a grant token or a GrantId/KeyId pair - have to set key_id.
+            key_id = store.grants[grant_id].metadata["KeyId"]
+
+        # KeyId can potentially hold one of multiple different types of key identifiers. _get_key finds a key no
+        # matter which type of id is used.
+        key_id = _get_key(key_id, store).metadata["KeyId"]
+        if grant_id not in store.grants:
+            raise InvalidGrantIdException()
+        if store.grants[grant_id].metadata["KeyId"] != key_id:
+            raise ValidationError(f"Invalid KeyId={key_id} specified for grant {grant_id}")
+
+        grant = store.grants[grant_id]
+        # In AWS grants have one or more tokens. But we have a simplified modeling of grants, where they have exactly
+        # one token.
+        store.grant_tokens.pop(grant.token)
+        store.grant_names.pop(grant.metadata.get("Name"), None)
+        store.grants.pop(grant_id)
+
+    @handler("RevokeGrant")
```

Comment:
nit: `@handler` decorator technically not required, as the identical decorator is already defined in the interface. (we usually only overwrite `@handler` if we pass custom attributes like `expand=False`)

(Same for the other decorators in the class)

Reply:
Done.